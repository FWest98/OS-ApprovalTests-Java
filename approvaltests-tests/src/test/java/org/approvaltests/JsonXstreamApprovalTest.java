package org.approvaltests;

import org.junit.jupiter.api.Test;

import java.time.Instant;

public class JsonXstreamApprovalTest
{
  @Test
  public void testBasicFormatting()
  {
    Circular c = new Circular();
    JsonXtreamApprovals.verifyAsJson(c);
  }

  public static class Circular {
    Circular parent;
    String label;

    public Circular() {
      this.parent = this;
      this.label = "Label";
    }
  }
}
