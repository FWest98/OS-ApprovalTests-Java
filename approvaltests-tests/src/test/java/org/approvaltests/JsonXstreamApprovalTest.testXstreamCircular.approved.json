{"org.approvaltests.Circular": {
  "parent": {
    "parent": {
      "@reference": "../.."
    },
    "label": "child"
  },
  "label": "middle"
}}